src/parameterizations/gravity_wave_drag/orographic_gravity_wave_helper.jl

```diff
@@ -3,12 +3,186 @@ using Interpolations
+        )
+    dAdy =
+        hcat(
+            A[:, 2] .- A[:, 1],
+            (A[:, 3:end] .- A[:, 1:(end - 2)]) / FT(2),
+            A[:, end] .- A[:, end - 1],
+        ) ./ (deg2rad(dlat) * earth_radius)
+    return (dAdx, dAdy)
+end
+
+"""
+    calc_velocity_potential(elev, lon, lat, earth_radius)
+    
+    Calculate velocity potential
+    - elev: surface elevation
+    - lon: longitude
+    - lat: latitude
+    - earth_radius: radius of the Earth 
+"""
+function calc_velocity_potential(elev, lon, lat, earth_radius)
+    FT = eltype(elev)
+    @. elev = max(0, elev)
+
+    dlat = lat[2] - lat[1]
+    dlon = lon[2] - lon[1]
+
+    scale =
+        sind(40) ./ earth_radius ./ sind.(max.(FT(20), abs.(lat))) .* FT(100e3) # FT(200e3) is used in the codes sent by Steve Garner
+
+    # compute weights for the spatial running mean using the Blackman kernel
+    ilat_range = Int.(round.(scale ./ deg2rad(dlat)))
+    ilon_range =
+        min.(
+            Int.(round.(scale ./ (deg2rad(dlon) .* cosd.(lat)))),
+            Int(round(length(lon) / 8)),
+        )
+
+    χ = zeros(size(elev))
+    for (i, ilon) in enumerate(lon)
+        for (j, jlat) in enumerate(lat)
+            # irange may not need clipping at the boundaries since it is on the closed lat circle
+            irange =
+                max(i - ilon_range[j], 1):min(i + ilon_range[j], length(lon))
+            jrange =
+                max(j - ilat_range[j], 1):min(j + ilat_range[j], length(lat))
+
+            # compute weights for spatial (area) integration
+            arc =
+                acos.(
+                    min.(
+                        FT(1),
+                        reshape(
+                            repeat(
+                                cosd(jlat) .* cosd.(lat[jrange]),
+                                length(irange),
+                            ),
+                            length(jrange),
+                            :,
+                        )' .* reshape(
+                            repeat(cosd.(ilon .- lon[irange]), length(jrange)),
+                            length(irange),
+                            :,
+                        ) .+
+                        reshape(
+                            repeat(
+                                sind(jlat) .* sind.(lat[jrange]),
+                                length(irange),
+                            ),
+                            length(jrange),
+                            :,
+                        )',
+                    )
+                )
+            arc1 = arc ./ max.(arc, scale[j])
+            wts =
+                reshape(
+                    repeat(cosd.(lat[jrange]), length(irange)),
+                    length(jrange),
+                    :,
+                )' ./ (arc .+ eps(FT)) .* (
+                    FT(0.42) .+ FT(0.50) .* cos.(pi * arc1) .+
+                    FT(0.08) .* cos.(FT(2) * pi * arc1)
+                )
+            if i != irange[1] && j != jrange[1]
```

Comment:
You probably don't need this line anymore. Or if you still need it, should it be `if i != irange[1]-1 && j != jrange[1]-1`?